### PR TITLE
Subnets API

### PR DESCRIPTION
**Pull request by Pajk**
API for subnets. It does not contain #import and #create_multiple methods from subnets controller which imho don't have to be in API.

### CODE REVIEW

File: app/controllers/api/v1/subnets_controller.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        param :name, String, :desc => 'Subnet name', :required => true
+        param :network, String, :desc => 'Subnet network', :required => true
+        param :mask, String, :desc => 'Netmask for this subnet', :required => true
+        param :gateway, String, :desc => 'Primary DNS for this subnet'
+        param :dns_primary, String, :desc => 'Primary DNS for this subnet'
+        param :dns_secondary, String, :desc => 'Secondary DNS for this subnet'
+        param :from, String, :desc => 'Starting IP Address for IP auto suggestion'
+        param :to, String, :desc => ' Ending IP Address for IP auto suggestion'
+        param :vlanid, String, :desc => 'VLAN ID for this subnet'
+        param :domain, Array, :desc => 'Domains in which this subnet is part'
```
Comment (ohadlevy) [inline app/controllers/api/v1/subnets_controller.rb]: isnt it domain_ids ?

File: app/views/api/v1/subnets/show.json.rabl [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+end
+
+child :tftp => :tftp do
+  attributes :id, :name, :url
+end
+
+child :dns => :dns do
+  attributes :id, :name, :url
+end
\ No newline at end of file
```
Comment (ohadlevy) [inline app/views/api/v1/subnets/show.json.rabl]: I'm missing the associated domains

Comment (Pajk) [inline app/views/api/v1/subnets/show.json.rabl]: Will this be enough info?

```
child :domains do
  attributes :id, :name, :fullname
end
```

File: app/controllers/api/v1/subnets_controller.rb (hunk 1 above)
Comment (Pajk) [inline app/controllers/api/v1/subnets_controller.rb]: Yes you are right.

File: app/views/api/v1/subnets/show.json.rabl (hunk 2 above)
Comment (ohadlevy) [inline app/views/api/v1/subnets/show.json.rabl]: yes, maybe it would be enough to return the domain ids ?

if not, I would use the domain partial instead

Comment (Pajk) [inline app/views/api/v1/subnets/show.json.rabl]: OK returning just domain_ids, for more info there is domain API available.

**Comment by ohadlevy** [comment]
merged - thanks!

**Pull request closed without merging**